### PR TITLE
Package name escaping

### PR DESCRIPTION
**Pull request by iankendrick**
### Description

This fixes an issue where the package resource would fail for packages with "+" in their name. This includes a few dozen csw packages (https://www.opencsw.org/get-it/packages/).

Changes made:
- quote and use safe() method when calling `pkgutil --parse`
- Added bonnie++ to test data
  (not sure what it is, but it's the lightest package I could find with a "+" in the name)
#### Bonus
- Fix kitchen vagrant tests. This can be rebased into a separate PR if required.
#### Notes
- Tests updated to use https://atlas.hashicorp.com/tnarik/boxes/solaris10-minimal as chef/solaris10 is no longer available
### Issues Resolved

No issue created
### Check List
- [*] All tests pass. See https://github.com/chef-cookbooks/community_cookbook_documentation/blob/master/TESTING.MD
- [*] New functionality includes testing.
- [N/A] New functionality has been documented in the README if applicable
- [*] The CLA has been signed. See https://github.com/chef-cookbooks/community_cookbook_documentation/blob/master/CONTRIBUTING.MD

### CODE REVIEW

File: .kitchen.yml [hunk 1]
```diff
@@ -18,10 +18,10 @@ platforms:
   - name: solaris-10.11
     run_list: pkgutil::opencsw
     driver:
-      box: chef/solaris-10.11 # private
+      box: tnarik/solaris10-minimal
```
Comment (tas50) [inline .kitchen.yml]: This needs to stay chef/solaris

Comment (iankendrick) [inline .kitchen.yml]: Okay, will revert. Do I need to build chef/solaris from bento and re-test?

**Comment by iankendrick** [comment]
Rebased to remove change to test kitchen basebox.  Box used is `chef/solaris-10.11` once again.

**Comment by tas50** [comment]
Thanks. This is a nice fix

**Pull request merged**